### PR TITLE
type: Loader.resource is a list of ILoaderResource

### PR DESCRIPTION
**Pull request by AdrienLemaire**
##### Description of change

The current implementation returns errors when trying to get a texture from a resource
```
[tsserver 2339] [E] Property 'texture' does not exist on type 'Resource | {}'
```
After reading the code, I found that pixi.js extends the resource-loader Resource type with the desired properties
```ts
export declare interface ILoaderResource extends GlobalMixins.ILoaderResource, Resource {
    texture?: Texture;
    spritesheet?: Spritesheet;
    textures?: Dict<Texture>;
    metadata: IResourceMetadata;
}
```

##### Pre-Merge Checklist

- [ ] Tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: packages/loaders/src/Loader.ts [hunk 1]
```diff
@@ -155,6 +155,15 @@ export class Loader extends ResourceLoader
...
         return Loader;
     }
+
+    /**
+     * All the resources for this loader keyed by name.
+     *
+     * @member {object<string, ILoaderResource>}
+     */
+    resources: {
+        [key: string]: import('@pixi/loaders').ILoaderResource;
```
Comment (ivanpopelyshev) [inline packages/loaders/src/Loader.ts:165]: Referencing ILoaderResource from module itself, are we sure just `ILoaderResource` wont work?

**Review by ivanpopelyshev — CHANGES_REQUESTED** [review]
Please remove package.json from this commit, its not your problem  to handle ;)

File: packages/loaders/src/Loader.ts (hunk 1 above)
Comment (bigtimebuddy) [reply packages/loaders/src/Loader.ts:165]: Please don't use this import syntax here. Import type at the start of the file to import the interface. 

I thought that resources was a property of resource-loader. Why is it not inherited?

**Comment by SukantPal** [comment]
@AdrienLemaire We don't commit scoped package-specific `package-lock.json` files into the repository.

File: packages/loaders/src/Loader.ts [hunk 2]
```diff
@@ -155,6 +155,15 @@ export class Loader extends ResourceLoader
 
         return Loader;
     }
+
+    /**
+     * All the resources for this loader keyed by name.
+     *
+     * @member {object<string, ILoaderResource>}
```
Comment (SukantPal) [inline packages/loaders/src/Loader.ts:162]: I know `PIXI.ILoaderResource` isn't documented, but all of the API is under `PIXI`. Maybe we should add docs to `PIXI.ILoaderResource` @bigtimebuddy?

```suggestion
     * @member {object<string, PIXI.ILoaderResource>}
```

**Review by SukantPal — CHANGES_REQUESTED** [review]

**Comment by bigtimebuddy** [comment]
Thanks for pointing this type error, @AdrienLemaire. Because we had to make some other fixes, I'm closing this in favor of #7300. I will credit you in the commit though.

**Comment by AdrienLemaire** [comment]
My pleasure.
@ivanpopelyshev for the package-lock.json, it would be helpful to add `package-lock.json` instead of `packages/**/package-lock.json` the gitignore
https://github.com/pixijs/pixi.js/blob/dev/.gitignore#L1

**Pull request closed without merging**